File: 1109.py
class Solution:
    def nextLargerNodes(self, head: ListNode) -> List[int]:
        elements = []
        pointer = head
        # convert the linkedlist to array
        while pointer:
            elements.append(pointer.val)
            pointer = pointer.next
        stack = []
        res = [0 for _ in range(len(elements))]
        index = 0
        while index < len(res):
            while stack and elements[stack[-1]] < elements[index]:
                res[stack.pop()] = elements[index]
            stack.append(index)
            index+=1
        return res
// should rememeber this method
// the "monotonic stack" is used for solve the problems like next greater/smaller element for both linkedlist/array

File: 1109.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public int[] nextLargerNodes(ListNode head) {
        List<Integer> elements = new ArrayList<>();
        ListNode pointer = head;
        while(pointer!=null){
            elements.add(pointer.val);
            pointer = pointer.next;
        }
        // monotonic stack
        Stack<Integer> stack = new Stack<>();
        int[] res = new int[elements.size()];
        for(int i = 0;i<elements.size();i++){
            while(stack.size()>0 && elements.get(stack.peek()) < elements.get(i))
                res[stack.pop()] = elements.get(i);
            stack.push(i);
        }
        return res;
    }
}
